File: Binary Search/Upper Bound/upper_bound.cpp
#include<bits/stdc++.h>
using namespace std;

int check(int arr[], int mid, int x) {
	if(arr[mid]>x) return 1;
	else return 0;
}

int main() {
	cout<<"Enter the size of the array: ";
	int n;
	cin>>n;
	int arr[n];
	cout<<"Enter "<<n<< " elements of the array: ";
	for(int i = 0;i<n;i++) cin>>arr[i];
	int x;
	cout<<"Enter the target element to find lower bound in the array: ";
	cin>>x;
	int low = 0;
    int high = n-1;
    int ans = n;
    while(low<=high) {
    	int mid = low+(high-low)/2;
    	if(check(arr, mid, x)==1) {
    		ans = mid;
    		high = mid-1;
    	} else {
    		low = mid+1;
    	}
    }
    cout<<ans<<endl;
	return 0;
}

File: Binary Search/Upper Bound/upper_bound.java
import java.util.Scanner;

public class Main {
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		System.out.println("Enter size of the array: ");
		int n = sc.nextInt();

		int[] arr = new int[n];
		System.out.println("Enter " + n + " elements of the array: ");
		for(int i = 0;i<n;i++) {
			arr[i] = sc.nextInt();
		}

		System.out.println("Enter the target element to find lower bound in the array: ");
		int x = sc.nextInt();

		int low = 0;
		int high = n-1;
		int ans = n;
		while(low<=high) {
			int mid = low+(high-low)/2;
			if(arr[mid]>x) {
				ans = mid;
				high = mid-1;
			} else {
				low = mid+1;
			}
		}
		System.out.println(ans);
	}
}